config/initializers/semantic_logger.rb

```diff
@@ -1,18 +1,65 @@
-if Settings.logstash.host && Settings.logstash.port
-  logstash_formatter = Proc.new do |event|
-    # The value here appears to break logging to logstash / elasticsearch
-    event["duration"] = event["duration_ms"]
-    event["duration_ms"] = nil
+class LogStashFormatter < SemanticLogger::Formatters::Raw
+  def format_json_message_context
+    if hash[:message].present?
+      context = JSON.parse(hash[:message])["context"]
+      hash[:sidekiq_job_context] = hash[:message]
+      hash[:message] = context
+    end
+  rescue JSON::ParserError
```

Comment:
Is this the right thing to do? I _think_ this executes in a separate thread and if this exception is raised it won't actually interfere with Sidekiq.

Reply:
The intent here is to continue as normal if the data in the message isn't valid JSON, rather than deciding based on whether or not it begins with a given string.

Reply:
My concern is just that we'll swallow up this error and never see it. OTOH, if we raise an error in this thread, I expect it'll log something to stdout and keep trying to send stuff to logstash. Will approve anyway as this is mostly conjecture.

Reply:
No the error will still be sent to logstash, it just won't be nicely formatted